geode-core/src/main/java/org/apache/geode/internal/InternalDataSerializer.java

```diff
@@ -855,13 +855,14 @@ static DataSerializer newInstance(Class c) {
       init = c.getDeclaredConstructor(new Class[0]);
 
     } catch (NoSuchMethodException ignored) {
-      String s = "Class %s does not have a zero-argument constructor.";
-      Object[] args = new Object[] {c.getName()};
       if (c.getDeclaringClass() != null) {
-        s =
+        String s =
             "Class %s does not have a zero-argument constructor. It is an inner class of %s. Should it be a static inner class?";
-        args = new Object[] {c.getName(), c.getDeclaringClass()};
+        Object[] args = new Object[] {c.getName(), c.getDeclaringClass()};
```

Comment:
String.format accepts String varargs, so you can get rid of the Object[] creations:
```
throw new IllegalArgumentException(String.format(s, c.getName(), c.getDeclaringClass()));
```